### ISSUE TITLE
Handling "fragment encoded redirect" to return URL from Identity Server 4

### ISSUE THREAD

**Issue by joshiji** [gsuitedevs/apps-script-oauth2#127]
Hi there.

First of all, thanks for providing this great library. I was able to use it with help of provided documentation.

I have only one open item with the usage. I am using an IdentityServer4 based authentication/authorization system and using this library to get tokens from that server. I am having only issue that server returns OAuth params in "fragment encoded redirect" instead of URI params. So, it has return URL something like:

`https://script.google.com/macros/d/[SCRIPT ID]/usercallback#code=`

So, I get a page not found error. When I replace the hash (#) in the URL with (?), my authCallBack function is hit and I am getting proper response. I looked into other issues like one with dropbox. Since I am using a Hybrid flow, I used following:

`.setParam('response_type', 'code id_token token')`

But, still the issue persists. Based on Identity Server docs at http://docs.identityserver.io/en/release/endpoints/authorize.html, I changed `response_mode` to `form_post` and token format to JSON. But, I am still not able to resolve it. Now, the data is not present in URL thus I believe it is being sent in JSON(form data).

Can you please help me in identifying what I am missing here?

**Comment by erickoledadevrel** [comment]
I just added a samples that shows how to use this library to connect to their demo API:

https://github.com/gsuitedevs/apps-script-oauth2/blob/master/samples/IdentityServer4.gs

I think the problem is that you are using the Hybrid flow, which by design uses the "browser channel" (AKA URL hash) to transmit the information. This works well for client-side application, but not as well for server-side ones like Apps Script. The Authorization Code flow is what this library typically expects and worked well. Note I was able to get the Hybrid flow working with the `form_post` option set, but it doesn't seem to offer any value over using the Authorization Code flow.
